core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -66,22 +70,32 @@
 import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static io.airlift.concurrent.Threads.threadsNamed;
 import static io.trino.execution.executor.MultilevelSplitQueue.computeLevel;
+import static io.trino.execution.executor.PrioritizedSplitRunner.SPLIT_RUN_QUANTA;
 import static io.trino.version.EmbedVersion.testingVersionEmbedder;
 import static java.lang.Math.min;
 import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+import static java.util.Arrays.asList;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.Executors.newCachedThreadPool;
 import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
 import static java.util.concurrent.TimeUnit.MICROSECONDS;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.joining;
 
 @ThreadSafe
 public class TaskExecutor
 {
     private static final Logger log = Logger.get(TaskExecutor.class);
 
     // print out split call stack if it has been running for a certain amount of time
-    private static final Duration LONG_SPLIT_WARNING_THRESHOLD = new Duration(600, TimeUnit.SECONDS);
+    private static final Duration LONG_SPLIT_WARNING_THRESHOLD = new Duration(600, SECONDS);
```

Comment:
We also need to make `LONG_SPLIT_WARNING_THRESHOLD` configurable and need to verify that `LONG_SPLIT_WARNING_THRESHOLD <= interruptStuckSplitsTimeout` (you can refer to #13042)

Reply:
@arhimondr How do you think, should we keep API /v1/maxActiveSplits and make the threshold configurable via config?